drivers/ethernet/eth_stm32_hal.c

```diff
@@ -953,6 +951,15 @@ static int eth_initialize(const struct device *dev)
 		LOG_ERR("Could not configure ethernet pins");
 		return ret;
 	}
+	return 0;
+}
+
+static int eth_post_initialize(void)
```

Comment:
Something like `eth_configure()` would also work and may be clearer.

Reply:
configure does not say anything about execution order, which is mandatory.

Reply:
Well these init/configure steps are very common in other drivers, with the notable difference that configure is generally called either within init or later through API call

Reply:
OK, the naming convention now reflect now to the current situation - the second part of the initialization is the configuration level.